From 3fbc984a8c73cb8245ea0c3783cb88c558996f7d Mon Sep 17 00:00:00 2001
From: Max Ihlenfeldt <max@igalia.com>
Date: Tue, 7 May 2024 17:38:25 +0200
Subject: [PATCH] Backport "gn bootstrap: Don't pass removed dupbuild ninja option"

This backports a patch to GN's bootstrap script to fix a build error
when using ninja >= 1.12.0. It will be included in upstream's 126
release.

Upstream-Status: Backport [https://crrev.com/c/5522398]
---
 tools/gn/bootstrap/bootstrap.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tools/gn/bootstrap/bootstrap.py b/tools/gn/bootstrap/bootstrap.py
index caf736b28946d..b721e79a205ed 100755
--- a/tools/gn/bootstrap/bootstrap.py
+++ b/tools/gn/bootstrap/bootstrap.py
@@ -122,7 +122,7 @@ def main(argv):

   shutil.copy2(
       os.path.join(BOOTSTRAP_DIR, 'last_commit_position.h'), gn_build_dir)
-  cmd = [ninja_binary, '-C', gn_build_dir, '-w', 'dupbuild=err', 'gn']
+  cmd = [ninja_binary, '-C', gn_build_dir, 'gn']
   if options.jobs:
     cmd += ['-j', str(options.jobs)]
   subprocess.check_call(cmd)
